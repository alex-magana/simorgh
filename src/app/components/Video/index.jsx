import React from 'react';
import { videoComponentPropTypes } from '../../models/propTypes';

const Video = ({
  pid,
  kind,
  title,
  items,
  holdingImageUrl,
  statsAppName,
  statsAppType,
  statsCountername,
  statsDestination,
  uiLocale,
}) => (
  <div>
    <div>video pid: {pid}</div>
    <div>kind: {kind}</div>
    <div>title: {title}</div>
    <div>holdingImageURL: {holdingImageUrl}</div>
    <div>items: {JSON.stringify(items, null, 4)}</div>
    <div>statsAppName: {statsAppName}</div>
    <div>statsAppType: {statsAppType}</div>
    <div>statsCountername: {statsCountername}</div>
    <div>statsDestination: {statsDestination}</div>
    <div>uiLocale: {uiLocale}</div>
  </div>
);

Video.propTypes = videoComponentPropTypes;

export default Video;
