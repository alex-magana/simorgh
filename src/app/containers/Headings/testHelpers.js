const blocksSingleFragment = (text, attributes = []) => [
  {
    type: 'text',
    model: {
      blocks: [
        {
          type: 'paragraph',
          model: {
            text,
            blocks: [
              {
                type: 'fragment',
                model: {
                  text,
                  attributes,
                },
              },
            ],
          },
        },
      ],
    },
  },
];

export default blocksSingleFragment;
