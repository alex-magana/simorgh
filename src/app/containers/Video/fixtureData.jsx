import React from 'react';
import PropTypes from 'prop-types';
import VideoContainer from '.';
import { RequestContextProvider } from '../../contexts/RequestContext';
import {
  captionBlock,
  videoClipGlobalGuidanceBlock,
  videoClipGlobalPortraitBlock,
  videoClipUkGuidanceBlock,
  videoClipNonUkBlock,
  audioClipGlobalGuidanceBlock,
  audioClipNonUkBlock,
  audioClipUkOnlyBlock,
  audioEpisodeGlobalBlock,
} from './helpers/fixtures';

const generateFixtureData = ({ platform, blocks }) => (
  <RequestContextProvider
    id="c0000000000o"
    isUK
    origin="https://www.test.bbc.co.uk"
    platform={platform}
    statsDestination="NEWS_PS_TEST"
    statsPageIdentifier="news.articles.c0000000000o.page"
  >
    <VideoContainer blocks={blocks} />
  </RequestContextProvider>
);

generateFixtureData.propTypes = {
  platform: PropTypes.string,
  blocks: PropTypes.arrayOf(PropTypes.any),
};

generateFixtureData.defaultProps = {
  platform: 'canonical',
  blocks: '',
};

export const NoData = ({ platform }) =>
  generateFixtureData({ platform, blocks: null });

export const NoAresMedia = ({ platform }) =>
  generateFixtureData({ platform, blocks: [captionBlock] });

export const VideoClipGlobalWithCaption = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [videoClipGlobalGuidanceBlock, captionBlock],
  });

export const VideoClipGlobalWithoutCaption = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [videoClipGlobalGuidanceBlock],
  });

export const VideoClipGlobalPortrait = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [videoClipGlobalPortraitBlock],
  });

export const VideoClipUkWithGuidance = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [videoClipUkGuidanceBlock],
  });

export const VideoClipNonUk = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [videoClipNonUkBlock],
  });

export const AudioClipGlobalGuidance = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [audioClipGlobalGuidanceBlock],
  });

export const AudioClipUk = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [audioClipUkOnlyBlock],
  });

export const AudioClipNonUk = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [audioClipNonUkBlock],
  });

export const AudioEpisodeGlobal = ({ platform }) =>
  generateFixtureData({
    platform,
    blocks: [audioEpisodeGlobalBlock],
  });
