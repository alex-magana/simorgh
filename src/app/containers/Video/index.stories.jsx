import { storiesOf } from '@storybook/react'; // eslint-disable-line
import {
  AudioClipGlobalGuidance,
  AudioClipNonUk,
  AudioClipUk,
  AudioEpisodeGlobal,
  VideoClipNonUk,
  VideoClipGlobalPortrait,
  VideoClipGlobalWithCaption,
  VideoClipUkWithGuidance,
  VideoClipGlobalWithoutCaption,
} from './fixtureData';
import AmpDecorator from '../../helpers/storybook/ampDecorator';

storiesOf('Video Container', module)
  .add('video, clip, with guidance, with caption', () =>
    VideoClipGlobalWithCaption({ platform: 'canonical' }),
  )
  .add('video, clip, with guidance, without caption', () =>
    VideoClipGlobalWithoutCaption({ platform: 'canonical' }),
  )
  .add('video, clip, UK, with guidance', () =>
    VideoClipUkWithGuidance({ platform: 'canonical' }),
  )
  .add('video, clip, non-UK, without guidance', () =>
    VideoClipNonUk({ platform: 'canonical' }),
  )
  .add('video, clip, global, without guidance, portrait', () =>
    VideoClipGlobalPortrait({ platform: 'canonical' }),
  )
  .add('audio, clip, global, with guidance', () =>
    AudioClipGlobalGuidance({ platform: 'canonical' }),
  )
  .add('audio, clip, UK, without guidance', () =>
    AudioClipUk({ platform: 'canonical' }),
  )
  .add('audio, clip, non-UK, without guidance', () =>
    AudioClipNonUk({ platform: 'canonical' }),
  )
  .add('audio, episode, global, without guidance', () =>
    AudioEpisodeGlobal({ platform: 'canonical' }),
  );

storiesOf('Video Container - AMP', module)
  .addDecorator(AmpDecorator)
  .add('video, clip, with guidance, with caption', () =>
    VideoClipGlobalWithCaption({ platform: 'amp' }),
  )
  .add('video, clip, with guidance, without caption', () =>
    VideoClipGlobalWithoutCaption({ platform: 'amp' }),
  )
  .add('video, clip, UK, with guidance', () =>
    VideoClipUkWithGuidance({ platform: 'amp' }),
  )
  .add('video, clip, non-UK, without guidance', () =>
    VideoClipNonUk({ platform: 'amp' }),
  )
  .add('video, clip, global, without guidance, portrait', () =>
    VideoClipGlobalPortrait({ platform: 'amp' }),
  )
  .add('audio, clip, global, with guidance', () =>
    AudioClipGlobalGuidance({ platform: 'amp' }),
  )
  .add('audio, clip, UK, without guidance', () =>
    AudioClipUk({ platform: 'amp' }),
  )
  .add('audio, clip, non-UK, without guidance', () =>
    AudioClipNonUk({ platform: 'amp' }),
  )
  .add('audio, episode, global, without guidance', () =>
    AudioEpisodeGlobal({ platform: 'amp' }),
  );
