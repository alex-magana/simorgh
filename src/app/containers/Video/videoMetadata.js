import deepGet from '../../helpers/json/deepGet';

const videoObject = 'VideoObject';

const videoMetadata = blocks => {
  const aresMediaBlocks = deepGet(['model', 'blocks'], blocks);
  const listContent = [];
  const metadata = {
    video: {
      '@list': listContent,
    },
  };

  if (!aresMediaBlocks || aresMediaBlocks.length < 1) {
    return null;
  }

  const aresMetaDataBlocks = aresMediaBlocks.filter(
    block => block.type === 'aresMediaMetadata',
  );

  aresMetaDataBlocks.forEach(block =>
    listContent.push({
      '@type': videoObject,
      name: deepGet(['model', 'title'], block),
      description: deepGet(['model', 'synopses', 'short'], block),
      duration: deepGet(['model', 'versions', [0], 'duration'], block),
      thumbnailUrl: `https://${deepGet(['model', 'imageUrl'], block)}`,
    }),
  );

  return listContent.length > 0 ? metadata : null;
};

export default videoMetadata;
