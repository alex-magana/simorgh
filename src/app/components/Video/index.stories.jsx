import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import Video from './index';

const props = {
  pid: 'p00a00a',
  kind: 'clip',
  title: 'Static Data Example',
  items: [
    {
      versionID: 'p00p00v',
      kind: 'clip',
      duration: 10,
    },
  ],
  holdingImageUrl: 'https://www.foo.bar/baz.png',
  statsAppName: 'news',
  statsAppType: 'responsive',
  statsCountername: 'news.articles.c0000000000o.page',
  statsDestination: 'NEWS_PS_TEST',
  uiLocale: 'en-GB',
};

storiesOf('Video', module).add('default', () => <Video {...props} />);
