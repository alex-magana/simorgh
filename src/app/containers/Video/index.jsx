import React from 'react';
import Helmet from 'react-helmet';
import Figure from '@bbc/psammead-figure';
import deepGet from '../../helpers/json/deepGet';
import Video from '../../components/Video';
import Caption from '../Caption';
import videoMetadata from './videoMetadata';

import {
  videoPropTypes,
  emptyBlockArrayDefaultProps,
} from '../../models/propTypes';
import { filterForBlockType } from '../../helpers/blockHandlers';
import { RequestContext } from '../../contexts/RequestContext';

const VideoContainer = ({ blocks }) => {
  const { platform, statsDestination, statsPageIdentifier } = React.useContext(
    RequestContext,
  );

  if (!blocks) {
    return null;
  }

  const aresMediaBlock = filterForBlockType(blocks, 'aresMedia');

  if (!aresMediaBlock) {
    return null;
  }

  const metadata = videoMetadata(aresMediaBlock);

  const captionBlock = filterForBlockType(blocks, 'caption');

  const nestedModel = deepGet(['model', 'blocks', 0, 'model'], aresMediaBlock);
  const kind = deepGet(['subType'], nestedModel);
  const pid = deepGet(['id'], nestedModel);
  const title = deepGet(['title'], nestedModel);
  const version = deepGet(['versions', 0], nestedModel);
  const duration = deepGet(['duration'], version);
  const versionID = deepGet(['versionId'], version);
  const holdingImageUrl = deepGet(
    ['blocks', 1, 'model', 'blocks', 0, 'model', 'locator'],
    aresMediaBlock.model,
  );
  const items = [
    {
      versionID,
      kind,
      duration,
    },
  ];

  return (
    <>
      {metadata ? (
        <Helmet>
          {
            <script type="application/ld+json">
              {JSON.stringify(metadata)}
            </script>
          }
        </Helmet>
      ) : null}
      <Figure>
        <Video
          pid={pid}
          kind={kind}
          title={title}
          items={items}
          holdingImageUrl={holdingImageUrl}
          statsAppName="news"
          statsAppType={platform === 'amp' ? 'amp' : 'responsive'}
          statsCountername={statsPageIdentifier}
          statsDestination={statsDestination}
          uiLocale="en-GB"
        />
        {captionBlock ? <Caption block={captionBlock} video /> : null}
      </Figure>
    </>
  );
};

VideoContainer.propTypes = videoPropTypes;

VideoContainer.defaultProps = emptyBlockArrayDefaultProps;

export default VideoContainer;
