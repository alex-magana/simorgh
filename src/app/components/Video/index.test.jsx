import React from 'react';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';
import Video from './index';

describe('Video', () => {
  describe('with data', () => {
    const props = {
      pid: 'p00a0a0m',
      kind: 'episode',
      title: 'Hello World',
      items: [
        {
          versionID: 'p00a0a0v',
          kind: 'episode',
          duration: 100,
        },
      ],
      holdingImageUrl: 'https://foo/bar/baz.png',
      statsAppName: 'news',
      statsAppType: 'responsive',
      statsCountername: 'news.articles.c0000000000o.page',
      statsDestination: 'NEWS_PS_TEST',
      uiLocale: 'en-GB',
    };

    shouldShallowMatchSnapshot(
      'should render the video with valid props',
      <Video {...props} />,
    );
  });
});
