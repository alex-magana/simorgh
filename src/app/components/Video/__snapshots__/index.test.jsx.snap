// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video with data should render the video with valid props 1`] = `
<div>
  <div>
    video pid: 
    p00a0a0m
  </div>
  <div>
    kind: 
    episode
  </div>
  <div>
    title: 
    Hello World
  </div>
  <div>
    holdingImageURL: 
    https://foo/bar/baz.png
  </div>
  <div>
    items: 
    [
    {
        "versionID": "p00a0a0v",
        "kind": "episode",
        "duration": 100
    }
]
  </div>
  <div>
    statsAppName: 
    news
  </div>
  <div>
    statsAppType: 
    responsive
  </div>
  <div>
    statsCountername: 
    news.articles.c0000000000o.page
  </div>
  <div>
    statsDestination: 
    NEWS_PS_TEST
  </div>
  <div>
    uiLocale: 
    en-GB
  </div>
</div>
`;
